File: docs/guides/javascript/src/searchWithOptionalFilters.ts
import { algoliasearch } from 'algoliasearch';

import type { OptionalFilters, SearchParams } from 'algoliasearch';

const labels: string[] = []; // A list of labels

const reduceLabelsToFilters = (_labels: string[]): OptionalFilters => {
  return []; // Implement your logic here
};

const client = algoliasearch('ALGOLIA_APPLICATION_ID', 'ALGOLIA_API_KEY');

const optionalFilters = reduceLabelsToFilters(labels);
const searchParams: SearchParams = {
  query: '<YOUR_SEARCH_QUERY>',
  optionalFilters: optionalFilters,
};

await client.searchSingleIndex({ indexName: 'indexName', searchParams: searchParams });

File: docs/guides/scala/src/main/scala/searchWithOptionalFilters.scala
import scala.concurrent.Future
import scala.concurrent.ExecutionContext.Implicits.global
import scala.concurrent.Await
import scala.concurrent.duration.Duration

import algoliasearch.api.SearchClient
import algoliasearch.config.*
import algoliasearch.extension.SearchClientExtensions
import algoliasearch.search.OptionalFilters.SeqOfOptionalFilters
import algoliasearch.search.SearchParamsObject

val labels: List[String] = List() // A list of labels

val reduceLabelsToFilters: Seq[String] => SeqOfOptionalFilters = _ => {
  SeqOfOptionalFilters(Seq()) // Implement your logic here
}

def searchWithOptionalFilters(): Future[Unit] = {
  val client = SearchClient(appId = "ALGOLIA_APPLICATION_ID", apiKey = "ALGOLIA_API_KEY")

  val optionalFilters = reduceLabelsToFilters(labels)
  val searchParams = SearchParamsObject(query = Some("<YOUR_SEARCH_QUERY>"), optionalFilters = Some(optionalFilters))

  Await.result(
    client
      .searchSingleIndex(
        indexName = "<YOUR_INDEX_NAME>",
        searchParams = Some(searchParams)
      )
      .map(_ => Future.unit),
    Duration(5, "sec")
  )
}
